common/djangoapps/third_party_auth/management/commands/tests/test_saml.py

```diff
@@ -416,57 +392,122 @@ def test_run_checks_slug_mismatches(self, mock_set_custom_attribute):
+            f'has SAML config (id={config.id}, slug=\'{config.slug}\') '
+            f'that does not match the provider\'s slug.'
+        )
+        self.assertIn(expected_info, output)
+        self.assertIn('Slug mismatches: 1', output)
+
+    def test_run_checks_null_configurations(self):
         """
         Test the --run-checks command identifies providers with null configurations.
         """
-        SAMLProviderConfigFactory.create(
+        provider = SAMLProviderConfigFactory.create(
             site=self.site,
             slug='null-provider',
             saml_configuration=None
         )
 
+        with mock.patch('common.djangoapps.third_party_auth.models.SAMLConfiguration.current',
+                        return_value=None):
+            output = self._run_checks_command()
+
+        expected_warning = (
+            f'[WARNING] Provider (id={provider.id}, name={provider.name}, '
+            f'slug={provider.slug}, site_id={provider.site_id}) '
+            f'has no direct SAML configuration and no matching default configuration was found.'
+        )
+        self.assertIn(expected_warning, output)
+        self.assertIn('Missing configs: 2', output)
+
+    def test_run_checks_null_config_id(self):
+        """
+        Test the --run-checks command identifies providers with configurations that have null IDs.
+        This tests the new logic that checks for default_config.id is None.
+        """
+        provider = SAMLProviderConfigFactory.create(
+            site=self.site,
+            slug='null-id-provider',
+            saml_configuration=None
+        )
+
+        # Mock config with id=None (simulates broken default config)
+        mock_config = mock.Mock()
+        mock_config.id = None
+
+        with mock.patch('common.djangoapps.third_party_auth.models.SAMLConfiguration.current',
+                        return_value=mock_config):
+            output = self._run_checks_command()
+
+        expected_warning = (
+            f'[WARNING] Provider (id={provider.id}, name={provider.name}, '
+            f'slug={provider.slug}, site_id={provider.site_id}) '
+            f'has no direct SAML configuration and no matching default configuration was found.'
+        )
+        self.assertIn(expected_warning, output)
+        self.assertIn('Missing configs: 2', output)
+
+    def test_run_checks_with_default_config(self):
+        """
+        Test the --run-checks command correctly handles providers with default configurations.
+        """
+        provider = SAMLProviderConfigFactory.create(
+            site=self.site,
+            slug='default-config-provider',
+            saml_configuration=None
+        )
+
+        default_config = SAMLConfigurationFactory.create(
+            site=self.site,
+            slug='default',
+            entity_id='https://default.example.com'
+        )
+
+        output = self._run_checks_command()
+
+        self.assertIn('Missing configs: 1', output)
```

Comment:
Assuming the following is accurate, you could add this test higher up in the file and then
you could note in comments where this goes from 1 to 2, that the setup data accounts for 1.
```
def test_run_checks_setup_test_data(self):
        """
        Test the --run-checks command against initial setup test data.
        """
        output = self._run_checks_command()

        # The setup data includes a saml provider with a missing config
        self.assertIn('Missing configs: 1', output)
```